corehq/util/log.py

```diff
@@ -243,12 +243,14 @@ def display_seconds(seconds):
 
 
 def with_progress_bar(iterable, length=None, prefix='Processing', oneline=True,
-                      stream=None, step=None):
+                      stream=None, step=timedelta(seconds=10)):
```

Comment:
Setting this later, as you did with `stream=None`, might be better than setting a default value to a mutable datatype.

Reply:
According to the [Python 3.7 docs](https://docs.python.org/3.7/library/datetime.html#available-types), `timedelta` (and most other `datetime` types) are immutable (grep for "Objects of these types are immutable.")

Reply:
That's cool!